File: classes01/cpp/closing.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void)
{
    Mat image = imread("../../data/images/closing.png",IMREAD_GRAYSCALE);

    int closingSize=10;
    Mat element = getStructuringElement(MORPH_ELLIPSE,Size(2*closingSize+1,2*closingSize+1),Point(closingSize,closingSize));

    Mat imageMorphClosed;
    morphologyEx(image,imageMorphClosed,MORPH_CLOSE,element);

    imshow("Original", image);
    imshow("Closed", imageMorphClosed);

    waitKey(0);
    destroyAllWindows();

    return EXIT_SUCCESS;
}

File: classes01/python/closing.py
import cv2

image=cv2.imread("../../data/images/closing.png",cv2.IMREAD_GRAYSCALE)

if image is None:
    print("文件读取失败")

closingSize=10
element = cv2.getStructuringElement(cv2.MORPH_ELLIPSE,(2*closingSize+1,2*closingSize+1),(closingSize,closingSize))

imageMorphClosed= cv2.morphologyEx(image,cv2.MORPH_CLOSE,element,iterations=1)

cv2.imshow("Origin", image)
cv2.imshow("Closed", imageMorphClosed)

cv2.waitKey(0)
cv2.destroyAllWindows()